changelogs/unreleased/8052-agent-cache-improve-naming.yml

```diff
@@ -0,0 +1,7 @@
+description: >
+  The retention policy of items in the agent cache can be set via
+  the `evict_after_creation` and `evict_after_last_access` parameters.
+change-type: minor
+destination-branches: [master]
+sections:
```

Comment:
We probably also need to add an `update-note` section regarding the change in behavior of the `timeout` and `for_version` parameters on the cache. And also a `deprecation-note` to say to people should migrate away from the `timeout` and `for_version` parameters.